File: clients/javascript/src/models/authTokenOut.ts
// this file is @generated

export interface AuthTokenOut {
    id: string;
    name: string;
    created: Date;
    updated: Date;
    expiry?: Date | null;
    metadata: { [key: string]: string };
    ownerId: string;
    scopes: string[];
    /** Whether this token is currently enabled. */
    enabled: boolean;
}

export const AuthTokenOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AuthTokenOut {
        return {
            id: object['id'],
            name: object['name'],
            created: new Date(object['created']),
            updated: new Date(object['updated']),
            expiry: object['expiry'] ? new Date(object['expiry']) : null,
            metadata: object['metadata'],
            ownerId: object['owner_id'],
            scopes: object['scopes'],
            enabled: object['enabled'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AuthTokenOut): any {
        return {
            'id': self.id,
            'name': self.name,
            'created': self.created,
            'updated': self.updated,
            'expiry': self.expiry,
            'metadata': self.metadata,
            'owner_id': self.ownerId,
            'scopes': self.scopes,
            'enabled': self.enabled,
        };
    }
}

File: clients/rust/src/models/auth_token_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AuthTokenOut {
    pub id: String,

    pub name: String,

    pub created_at: jiff::Timestamp,

    pub updated_at: jiff::Timestamp,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub expiry: Option<jiff::Timestamp>,

    pub metadata: std::collections::HashMap<String, String>,

    pub owner_id: String,

    pub scopes: Vec<String>,

    /// Whether this token is currently enabled.
    pub enabled: bool,
}

impl AuthTokenOut {
    pub fn new(
        id: String,
        name: String,
        created_at: jiff::Timestamp,
        updated_at: jiff::Timestamp,
        metadata: std::collections::HashMap<String, String>,
        owner_id: String,
        scopes: Vec<String>,
        enabled: bool,
    ) -> Self {
        Self {
            id,
            name,
            created_at,
            updated_at,
            expiry: None,
            metadata,
            owner_id,
            scopes,
            enabled,
        }
    }

    pub fn with_expiry(mut self, value: impl Into<Option<jiff::Timestamp>>) -> Self {
        self.expiry = value.into();
        self
    }
}
